.travis.yml

```diff
@@ -1,5 +1,42 @@
 dist: trusty
+sudo: required
+
+matrix:
+    include:
+        - os: linux
+        - env: OS_TYPE=fedora
+        - env: INSTALL_REQUIREMENTS="dnf repolist; dnf install -y perl shellcheck devscripts sudo wget"
+
+        - os: linux
+        - env: OS_TYPE=opensuse
+        - env: INSTALL_REQUIREMENTS="zypper refresh; zypper in -y perl shellcheck devscripts sugo wget"
+
+        - os: linux
+        - env: OS_TYPE=dauchida/freebsd
+        - env: INSTALL_REQUIREMENTS="pkg install -y perl shellcheck devscripts sudo wget"
+
+        - os: linux
+        - env: OS_TYPE=debian:11
+        - env: INSTALL_REQUIREMENTS="sudo apt-get update -qq; sudo apt-get upgrade -y; sudo apt-get install -y -qq perl shellcheck devscripts sudo wget"
+
+        - os: linux
+        - env: OS_TYPE=ubuntu:22.04
+        - env: INSTALL_REQUIREMENTS="sudo apt-get update -qq; sudo apt-get upgrade -y; sudo apt-get install -y -qq perl shellcheck devscripts sudo wget"
+
+        - os: osx
+        - env: OS_TYPE=osx
+        - env: INSTALL_REQUIREMENTS="brew update; brew install perl shellcheck devscripts sudo wget"
+
 before_install:
-- sudo apt-get update -qq
-- sudo apt-get install -qq perl shellcheck devscripts
-script: prove
+    - [[ $TRAVIS_OS_NAME == "linux" ]] && docker pull ${OS_TYPE}
+
+install:
+    - [[ $TRAVIS_OS_NAME == "osx" ]] && docker run -d $OS /bin/sh -c "$INSTALL_REQUIREMENTS"
```

Comment:
Which value will be set for `$OS` variable?
If it is a linux distribution name (such as `ubuntu:22.04`),  we don't need to run on`osx` env, which cost more credits than usual.

Reply:
And, there is absolutely 100% no way I'm ever going to support MacOS. I spent weeks messing about with it several years ago, and it's not worth it (and the days of MacOS on a server are over, if they ever existed...they didn't really, but Apple did used to make servers). Besides that, the best "package manager" on MacOS is allegedly Homebrew, which is a disaster, especially for servers.